src/game/Inventory.cpp

```diff
@@ -1088,16 +1088,18 @@ void CheckForInventoryReplaceMe(Entity * io, Entity * old) {
  */
 bool TakeFromInventory(const Vec2s & pos) {
 	Entity * io = GetFromInventory(pos);
-	
-	if(io == NULL)
+
+	if (io == NULL)
 		return false;
-	
-	if(g_secondaryInventoryHud.dragEntity(io, pos))
-		return true;
-	
-	g_playerInventoryHud.dragEntity(io, pos);
 
-	return true;
+	bool res;
+
+	res = g_secondaryInventoryHud.dragEntity(io, pos);
```

Comment:
Instread of adding a variable, I think an early return would be appropriate here.